traffic_ops/app/lib/API/Job.pm

```diff
@@ -181,7 +185,7 @@ sub create_current_user_job {
 			return $self->forbidden("Forbidden. Delivery-service tenant is not available to the user.");
 		}
 	} else {
-		if ( !&is_oper($self) && !$self->is_delivery_service_assigned($ds_id) ) {
+		if ( !&is_portal($self) && !$self->is_delivery_service_assigned($ds_id) ) {
```

Comment:
this would never resolve to true given the condition on line 157, right? i think i'd rather see this rewritten:

```
if ( $tenant_utils->use_tenancy() ) {
		if ( !$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id) ) {
			return $self->forbidden("Forbidden. Delivery-service tenant is not available to the user.");
		}
	} else {
		if ( !&is_oper($self) && !$self->is_delivery_service_assigned($ds_id) ) {
			return $self->forbidden();
		}
	}
```

to

```
		if ( !$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id) ) {
			return $self->forbidden("Forbidden. Delivery-service tenant is not available to the user.");
		}
```

Reply:
I can't do that, since it would change the API behavior beyond the scope of this security fix. I shouldn't be checking for Role twice, though.

Reply:
i feel like you can simply live with this change alone:

```
if (!&is_portal($self)) {
	return $self->forbidden();
}
```

because `$tenant_utils->use_tenancy()` never evaluates to false anymore so that change to line 188 does nothing.

Reply:
That's true... so why is it still a mutable, global parameter?

Reply:
i thought there was some work done to completely remove the ability to turn off tenancy but i guess there were many pieces that were missed. so imo i'd take this opportunity to remove that `$tenant_utils->use_tenancy()`

Reply:
The `use_tenancy` parameter might still exist in the global profile, but there shouldn't be any code that actually checks that parameter anymore. Nobody should be adding code that checks that parameter, either. In the Perl, pretty much all the code that checked tenancy used that `$tenant_utils->use_tenancy()` parameter, so I decided to just make that function always return true instead of removing all the places in the Perl that the `use_tenancy()` function was called. I would be -1 on going back and removing all the code in the Perl that checks the `use_tenancy()` function currently, since that would not be worth doing now.

Reply:
Either way, I think it falls outside the scope of this bug fix

Reply:
> I would be -1 on going back and removing all the code in the Perl that checks the use_tenancy() function currently, since that would not be worth doing now.

sounds good